cmd/kubeadm/app/util/apiclient/wait.go

```diff
@@ -84,31 +84,58 @@ type controlPlaneComponent struct {
 	url  string
 }
 
+const (
+	// TODO: switch to /livez once all components support it
+	// and delete the endpointHealthz constant.
+	// https://github.com/kubernetes/kubernetes/issues/118158
+	endpointHealthz = "healthz"
+	endpointLivez   = "livez"
+)
+
 // getControlPlaneComponents takes a ClusterConfiguration and returns a slice of
-// control plane components and their secure ports.
+// control plane components and their health check URLs.
 func getControlPlaneComponents(cfg *kubeadmapi.ClusterConfiguration) []controlPlaneComponent {
-	portArg := "secure-port"
+	const (
+		portArg        = "secure-port"
+		addressArg     = "bind-address"
+		defaultAddress = "127.0.0.1"
```

Comment:
not in ipv6 , that should be "::1"

Reply:
by default KCM and scheduler are deployed with --bind-address=127.0.0.1 user can override. later in the function the override is checked.

apiserver is deployed without --bind-adress but with --advertise-address, that's where the probes in the manifest poke at too. in the latest commit added code to respect that.